### PR TITLE
Few fixes and updates to spells

### PR DESCRIPTION
**Pull request by raymondtfr**
Commits are pretty descriptive now  :stuck_out_tongue_winking_eye:

### CODE REVIEW

File: data/spells/scripts/attack/lightest_magic_missile_rune.lua [hunk 1]
```diff
@@ -4,7 +4,7 @@ combat:setParameter(COMBAT_PARAM_EFFECT, CONST_ME_ENERGYAREA)
 combat:setParameter(COMBAT_PARAM_DISTANCEEFFECT, CONST_ANI_ENERGY)
 
 function onGetFormulaValues(player, level, maglevel)
-	return -3, -5
+	return -3, -7
```
Comment (ranisalt) [inline data/spells/scripts/attack/lightest_magic_missile_rune.lua]: Please add sources for formula edits.

Comment (raymondtfr) [inline data/spells/scripts/attack/lightest_magic_missile_rune.lua]: It was my own research 😱 .

Comment (wgrriffel) [inline data/spells/scripts/attack/lightest_magic_missile_rune.lua]: it's pretty accurate to what tibia wikia says tho, it says around 5 damage

Comment (raymondtfr) [inline data/spells/scripts/attack/lightest_magic_missile_rune.lua]: Unfortunately, we can't really much on *wikia* on these kind of stuff. I learned much following/reading Gorak first researches of spells from vanilla, so I'm much involved on this aspect of this game xD.

The current constants (formulas) for these spells were changed when I did the same research again on Feb. 2017. I was waiting for the right moment to use them here.

By the way, we can't even the spell Practice Magic Missile Rune on vanilla anymore, weird xD.

**Comment by raymondtfr** [comment]
I have squashed a few other similar PRs into this one. There are still a few more things I could do to improve this, probably a WIP.

@ranisalt @djarek

**Comment by ranisalt** [comment]
Are you done, @raymondtfr ?

**Comment by raymondtfr** [comment]
@ranisalt This is all... I think.

**Comment by raymondtfr** [comment]
Should I remove [this](https://github.com/otland/forgottenserver/pull/2285/commits/f72242b2cf93a3e0981e568c37c9073845b4616a#diff-e48f380da2d4a71fa7398be673c8a435R379) redundant check? since ``item:decay()`` already does this.

@ranisalt

**Comment by djarek** [comment]
@raymondtfr 
Please try to submit mostly unrelated commits in separate PRs... It's a lot easier to review the changes.

**Comment by raymondtfr** [comment]
Is it because of ad25ba1? If so, I had to move that code in this same PR because it's been changed already in previous commits (676f837, 4a9a170, and f72242b).

If that isn't appropriate then I can submit a new PR with that commit once this one is merged.

**Comment by djarek** [comment]
@raymondtfr 
```
Files changed: 101
```
I'd say it's enough changes for one PR :D.

**Comment by raymondtfr** [comment]
Ok, I'll remove that commit. Once this PR gets merged I'll make another PR with that change. But there are still some things I am going to update, so should I do it on another PR or on this one still?

**Comment by djarek** [comment]
If you need to push a commit to fix something you introduced in this PR, go ahead, but try to leave "new" changes out. And create other PRs for them.

**Comment by raymondtfr** [comment]
WTH lol.

File: data/spells/lib/spells.lua [hunk 2]
```diff
@@ -309,42 +309,53 @@ function Player:addPartyCondition(combat, variant, condition, baseMana)
 		return false
 	end
 
+	local positions = {}
+	function onTargetTile(creature, position)
+		positions[#positions + 1] = position
+	end
+
+	combat:setCallback(CALLBACK_PARAM_TARGETTILE, "onTargetTile")
```
Comment (raymondtfr) [inline data/spells/lib/spells.lua]: I don't know enough to understand such particularities, but is this a good idea (redefining the tile callback function and calling ``setCallback`` each time such spells are cast.) The main reason I did this is to remove common duplicated code from the four spells (3a5a167).

@djarek

**Comment by ranisalt** [comment]
Thanks! Should you change anything from this PR please open a new one.

**Pull request merged**